src/main/323C.c

```diff
@@ -10821,25 +10821,155 @@ INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_800D34F0);
 
 INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_800D3510);
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_800D354C);
+void func_800D354C(struct UnkObj* arg0)
+{
+    s32* addr_801F3000 = (s32*)0x801F3000;
+    s32 temp_v1;
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_800D35D0);
+    arg0->unk40 = 0x1F00;
+    temp_v1 = *addr_801F3000;
+    arg0->unk14 = -1;
+    arg0->unk3C = temp_v1 + (s32)addr_801F3000;
+    arg0->unk15 = 0;
+    if (arg0->y_pos.i.hi == 0x10) {
+        arg0->unk42 = 0x7802;
+    } else {
+        arg0->unk42 = 0x7800;
+    }
+    arg0->x_pos.i.hi = 0xA0;
+    arg0->unk16 = 0;
+    arg0->unk47 = arg0->unk2;
+    arg0->unk4++;
+    if (arg0->unk2 < 9) {
+        arg0->unk4++;
+    }
+}
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_800D36AC);
+void func_800D35D0(struct UnkObj* arg0)
+{
+    s32* addr_801F3000 = (s32*)0x801F3000;
+    s32* addr_801F3008 = (s32*)0x801F3008;
+    u32 temp_v1;
 
-INCLUDE_ASM("asm/us/main/nonmatchings/323C", func_800D3700);
+    arg0->unk40 = 0x1E00;
+    arg0->unk30 = D_8010ECD4;
+    temp_v1 = *addr_801F3008;
+    arg0->unk14 = -1;
+    arg0->unk3C = temp_v1 + (s32)addr_801F3000;
+    arg0->unk15 = 0;
+    if (arg0->unk2 == -1) {
+        arg0->unk42 = 0x7806;
+        arg0->y_pos.i.hi = *(u8*)(D_80141BDF[0] * 2 + arg0->unk50) + 8;
```

Comment:
I'm thinking it's like this?
```
        arg0->y_pos.i.hi = arg0->unk50[D_80141BDF[0] * 2] + 8;
```
https://decomp.me/scratch/5DRlV

Same for func_800D38A0